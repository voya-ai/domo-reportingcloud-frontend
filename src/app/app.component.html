<div class="container-fluid">
  <div class="row">
    <nav class="col-sm-3 col-md-2 d-none d-sm-block bg-light sidebar">
      <ul class="nav nav-pills flex-column">
        <li class="nav-item">
          <a routerLink="/"
              routerLinkActive="active"
              class="nav-link">View reports</a>
        </li>
        <li class="nav-item">
          <a routerLink="/report/add"
              routerLinkActive="active"
              class="nav-link">Add report</a>
        </li>
      </ul>
    </nav>
  </div>
</div>

<main class="col-sm-9 ml-sm-auto col-md-10 pt-3" role="main">
  <router-outlet></router-outlet>
</main>
