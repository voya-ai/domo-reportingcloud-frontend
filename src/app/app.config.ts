import { OpaqueToken } from "@angular/core";
import { environment } from 'environments/environment';

export let APP_CONFIG = new OpaqueToken("app.config");

export interface IAppConfig {
    apiEndpoint: string;
}

export const AppConfig: IAppConfig = {
    apiEndpoint: environment.apiUrl
};
