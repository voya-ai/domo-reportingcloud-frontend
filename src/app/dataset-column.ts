export interface DatasetColumn {
  name: string,
  checked: boolean
}
