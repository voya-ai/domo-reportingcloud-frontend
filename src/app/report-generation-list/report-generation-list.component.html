<h2>Generated reports</h2>

<mat-list>
  <mat-list-item
    *ngFor="let report of reports">
    <h3 matLine>
      <a [href]="report.url" target="_blank">
        {{ report.name }}
      </a>
    </h3>
    <p matLine>
      <span> {{report.slug}} </span>
      <span><em> ({{report.status}}) </em></span>
    </p>
  </mat-list-item>
</mat-list>
