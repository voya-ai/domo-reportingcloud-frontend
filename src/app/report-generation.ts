export interface ReportGeneration {
  name: string;
  slug: string;
  status: string;
}
