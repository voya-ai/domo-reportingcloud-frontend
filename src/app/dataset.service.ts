import { Injectable, Inject } from '@angular/core';
import { Http } from '@angular/http';
import { Observable } from 'rxjs/Observable';
import { APP_CONFIG, IAppConfig } from './app.config';
import { DatasetColumn } from './dataset-column';

import 'rxjs/add/operator/map';
import 'rxjs/add/operator/filter';
import 'rxjs/add/operator/first';
import 'rxjs/add/operator/shareReplay';

@Injectable()
export class DatasetService {
  private data : Observable<any[]>;

  constructor(
    private _http: Http,
    @Inject(APP_CONFIG) private config
  ) {
    this.data = this._http.get(`${this.config.apiEndpoint}/datasets`)
      .map(res => res.json())
      .shareReplay(1);
  }

  search(term: string) : Observable<any[]> {
    return this.data
      .map
      (data => {
        return data.filter(item => {
          return item.name.search(new RegExp(term, "i")) !== -1;
        });
      });
  }

  get(id: string) : Observable<any> {
    return this._http.get(`${this.config.apiEndpoint}/datasets/${id}`)
      .map(res => res.json());
  }

  getColumns(id: string) : Observable<DatasetColumn[]> {
    return this.get(id)
      .map(res => res.schema.columns.map(column => {
        return {
          name: column.name,
          headerChecked: false,
          contentChecked: false
        }
      })).map(data => {
        return data.sort((a, b) => {
          if(a.name < b.name) return -1;
          if(a.name > b.name) return 1;
          return 0;
        });
      });
  }

}
