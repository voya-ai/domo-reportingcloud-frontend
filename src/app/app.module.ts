import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { HttpModule } from '@angular/http';

import { RouterModule, Routes } from '@angular/router';

import {
  MatAutocompleteModule,
  MatInputModule,
  MatCardModule,
  MatGridListModule,
  MatCheckboxModule,
  MatButtonModule,
  MatDatepickerModule,
  MatNativeDateModule,
  MatSelectModule,
  MatMenuModule,
  MatListModule,
  MatToolbarModule,
  MatProgressSpinnerModule
} from '@angular/material';

import { AppComponent } from './app.component';

import { APP_CONFIG, AppConfig } from './app.config';

import { ReportService } from './report.service';
import { DatasetService } from './dataset.service';
import { TemplateService } from './template.service';
import { ReportGenerationService } from './report-generation.service';

import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { ReportListComponent } from './report-list/report-list.component';
import { ReportAddComponent } from './report-add/report-add.component';
import { ReportDetailComponent } from './report-detail/report-detail.component';
import { ReportDatasetAutocompleteComponent } from './report-dataset-autocomplete/report-dataset-autocomplete.component';
import { ReportTemplateAutocompleteComponent } from './report-template-autocomplete/report-template-autocomplete.component';
import { ReportGenerationListComponent } from './report-generation-list/report-generation-list.component';
import { ReportsComponent } from './reports/reports.component';
import { ReportDetailSampleComponent } from './report-detail-sample/report-detail-sample.component';
import { ReportEditComponent } from './report-edit/report-edit.component';
import { DatasetColumnSelectComponent } from './dataset-column-select/dataset-column-select.component';

const appRoutes: Routes = [
    {
        path: 'report/add',
        component: ReportAddComponent
    },
    {
        path: 'report/:slug',
        component: ReportDetailComponent
    },
    {
        path: '',
        component: ReportsComponent,
        pathMatch: 'full'
    }
];

@NgModule({
  declarations: [
    AppComponent,
    ReportListComponent,
    ReportAddComponent,
    ReportDetailComponent,
    ReportDatasetAutocompleteComponent,
    ReportTemplateAutocompleteComponent,
    ReportGenerationListComponent,
    ReportsComponent,
    ReportDetailSampleComponent,
    ReportEditComponent,
    DatasetColumnSelectComponent
  ],
  imports: [
    BrowserModule,
    FormsModule,
    HttpModule,
    ReactiveFormsModule,
    NoopAnimationsModule,
    RouterModule.forRoot(appRoutes),
    MatAutocompleteModule,
    MatInputModule,
    MatCardModule,
    MatGridListModule,
    MatCheckboxModule,
    MatButtonModule,
    MatDatepickerModule,
    MatNativeDateModule,
    MatSelectModule,
    MatMenuModule,
    MatListModule,
    MatToolbarModule,
    MatProgressSpinnerModule
  ],
  providers: [
      { provide: APP_CONFIG, useValue: AppConfig },
      ReportService,
      DatasetService,
      TemplateService,
      ReportGenerationService
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
