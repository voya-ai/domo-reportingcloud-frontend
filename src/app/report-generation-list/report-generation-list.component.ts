import { Component, OnInit } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import { environment } from 'environments/environment';

import { ReportGenerationService } from 'app/report-generation.service';

import { ReportGeneration } from 'app/report-generation';

@Component({
  selector: 'app-report-generation-list',
  templateUrl: './report-generation-list.component.html',
  styleUrls: ['./report-generation-list.component.css']
})
export class ReportGenerationListComponent implements OnInit {
  reports: any[];

  constructor(private _reportService: ReportGenerationService) {
    this._reportService.get()
      .toPromise()
      .then(function(reports: ReportGeneration[]) {
        this.reports = reports
          .map(function(report : any) {
            report.url = `${environment.bucketUrl}${report.name}`;
            return report;
          });
      }.bind(this));
  }

  ngOnInit() {
  }
}
