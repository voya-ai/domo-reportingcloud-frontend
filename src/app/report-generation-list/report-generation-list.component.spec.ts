import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import {
  MatListModule
} from '@angular/material';
import { Observable } from 'rxjs/Observable';

import { ReportGeneration } from 'app/report-generation';
import { ReportGenerationService } from 'app/report-generation.service';

import { ReportGenerationListComponent } from './report-generation-list.component';

class MockReportGenerationService {
    get() : Observable<ReportGeneration[]> {
        return Observable.of<ReportGeneration[]>([]);
    }
}

describe('ReportGenerationListComponent', () => {
  let component: ReportGenerationListComponent;
  let fixture: ComponentFixture<ReportGenerationListComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      providers: [
          { provide: ReportGenerationService, useClass: MockReportGenerationService }
      ],
      declarations: [ ReportGenerationListComponent ],
      imports: [ MatListModule ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ReportGenerationListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
