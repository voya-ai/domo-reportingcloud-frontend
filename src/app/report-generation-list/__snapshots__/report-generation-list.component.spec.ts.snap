// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportGenerationListComponent should match the snapshot 1`] = `
<app-report-generation-list
  _reportService={[Function MockReportGenerationService]}
  reports={[Function Array]}
>
  <h2>
    Generated reports
  </h2>
  
  <mat-list
    class="mat-list"
    role="list"
  >
    
    
    
    
  
  </mat-list>
  
</app-report-generation-list>
`;
